link-grammar/wordgraph.c

```diff
@@ -197,15 +197,11 @@ bool wordgraph_pathpos_add(Wordgraph_pathpos **wp, Gword *p, bool used,
 		}
 	}
 
-	*wp = wordgraph_pathpos_resize(*wp, n+1);
-
+	*wp = wordgraph_pathpos_resize(*wp, n);
```

Comment:
the memove fix below is correct, but I don't believe that the above is. That's because I also changed the argument to `wordgraph_pathpos_resize` so that it expects the new length, not counting the terminating null. by contrast, the `n` is the old length.  You need to add one to make room for the new element.